python/ql/src/semmle/python/Concepts.qll

```diff
@@ -293,6 +293,78 @@ module SqlExecution {
   }
 }
 
+/**
+ * A data-flow node that escapes meta-characters, which could be used to prevent
+ * injection attacks.
+ *
+ * Extend this class to refine existing API models. If you want to model new APIs,
+ * extend `Escaping::Range` instead.
+ */
+class Escaping extends DataFlow::Node {
+  Escaping::Range range;
+
+  Escaping() {
+    this = range and
+    // escapes that don't have _both_ input/output defined are not valid
+    exists(range.getAnInput()) and
+    exists(range.getOutput())
```

Comment:
I'm not quite sure what the function of these two lines is. What would go wrong if there was a "half-defined" instance of `Escaping`?

Reply:
The problematic part is that we do `override predicate isSanitizer(DataFlow::Node node) { node = any(HtmlEscaping esc).getOutput() }` in the XSS configuration, and then the example from the commit message of https://github.com/github/codeql/pull/6092/commits/498703fc81727d733265dc0c9002205cfd8b5534